components/TabBar/selectors.js

```diff
@@ -56,9 +62,13 @@ export const getActiveTab = createSelector(
  * @returns {boolean}
  */
 export const isTabBarVisible = createSelector(
+  getTabBarState,
   getHistoryPathname,
   hasFavorites,
-  (pathname, favorites) => {
+  (state, pathname, favorites) => {
+    if (state.show === false) {
```

Comment:
With this implementation it will be only possible to hide the `TabBar` on pages where it's supposed to be shown. But if we would like to show it on pages where it's usually hidden, we couldn't use this system.

So it think that it would make more sense to move the page detection logic into the stream system and dispatch the `showTabBar` and `hideTabBar` actions from there. The selector would return the current value of the `show` property.

Reply:
I also think it makes sense, but I think the effort that is required for it is too high.

Currently there are two, documented conditions for those components to be visible or hidden. One is the redux state (dynamic), second is the current pathname (static).

Both must allow the components to be visible and I'm fine with it. 

I'd name those actions differently though (allow/disallowTabBarVisible...) but when talked about it with Destiny, they said, show/hide is better.

Reply:
I actually fixed the TabBar tests.